View/ViewHandler.php

```diff
@@ -448,6 +449,18 @@ private function initResponse(View $view, $format)
             if ($serializer instanceof SerializerInterface) {
                 $context = $this->getSerializationContext($view);
                 $content = $serializer->serialize($data, $format, $context);
+            //       test for symfony 2.7 version 
+            } elseif (class_exists('\Symfony\Component\Serializer\Normalizer\ObjectNormalizer') && $serializer instanceof SymfonySerializerInterface) {
+                    
+                $context = $this->getSerializationContext($view);
+
+                $newContext = array();
+
+                if (!$context->attributes->get('groups')->isEmpty()) {
+                    $newContext['groups'] = $context->attributes->get('groups')->get();
+                }
+
+                $content = $serializer->serialize($data, $format, $newContext);
```

Comment:
why not just move all of this code into the else clause? it should not hurt to pass an extra parameter